### PR TITLE
Move decorator location

### PR DESCRIPTION
**Pull request by adrianschmidt**
Should not be rebased into `alpha`. I'll bring this into `alpha` manually once approved.

The review is regarding the commits _after_ the merge.

### CODE REVIEW

File: packages/lime-web-components/src/index.ts [hunk 1]
```diff
@@ -17,3 +17,4 @@ export * from './device';
 export * from './filter';
 export * from './userdata';
 export * from './application';
+export * from './location';
```
Comment (jgroth) [inline packages/lime-web-components/src/index.ts:20]: I think this should live together with the other navigation stuff in the `navigator` folder

File: packages/lime-web-components/src/location/decorator.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+/**
+ * Gets a named query param from the location
+ *
+ * @experimental
+ * @param {StateOptions} options state decorator options
+ * @returns {PropertyDecorator} state decorator
+ */
+export function QueryParam(options: LocationOptions): PropertyDecorator {
```
Comment (jgroth) [inline packages/lime-web-components/src/location/decorator.ts:19]: For consistency, I think we should rename this to `SelectQueryParam`

File: packages/lime-web-components/src/index.ts (hunk 1 above)
Comment (adrianschmidt) [reply packages/lime-web-components/src/index.ts]: ⚡

File: packages/lime-web-components/src/location/decorator.ts (hunk 2 above)
Comment (adrianschmidt) [reply packages/lime-web-components/src/location/decorator.ts]: ⚡

I also renamed `LocationOptions` to `SelectQueryParamOptions` in a separate fixup.

**Comment by adrianschmidt** [comment]
Superseded by #434

**Pull request closed without merging**